src/components/new-request/search.vue

```diff
@@ -102,9 +102,7 @@
       <!-- Change Name flow -->
       <template v-else-if="isChangeName">
         <BusinessLookupFetch />
-        {{ isSocietyDisabled }}{{ showSocietiesInfo }}
-        <!--  :show-modal="isSocietyDisabled && showSocietiesInfo"-->
-        <SocietiesInfo type='name' />
+        <SocietiesInfo v-if="isSocietyDisabled && showSocietiesInfo" :type="'request a name for'" :showDialog="showSocietiesInfo"/>
```

Comment:
You were asking about best practises for modals...

In this repo, all dialogs (except this one) are declared in App.vue and each one uses its own store getter + action to show/hide. Personally I don't like this approach because a lot of code files are involved; it's not "contained" or localized.

In the other repos, each dialog generally looks like this in the parent component:
```
<MyDialog
  :dialog="whetherToShowThisDialog"
  @close="closeThisDialog()"
/>

<script>
  ...
  whetherToShowThisDialog = false // initial value
  ...
  showThisDialog (): void {
    this.whetherToShowThisDialog = true
  }
  ...
  closeThisDialog (): void {
    this.whetherToShowThisDialog = false
  }
  ...
```

I prefer this approach because I don't have to touch any other system files (App.vue, interfaces, store, state, getters, actions). It's all here. And the parent can respond to close events (or you  could implement "exit" or "retry" events instead).

What do you think? There isn't a correct answer but depending on the app and what you're doing in it, there may be a "better" answer :)

Reply:
Example: https://github.com/bcgov/business-filings-ui/blob/05b5e3b9950dd5d2f598c66d008394eca99466a5/src/views/AgmLocationChg.vue#L8

Reply:
You're right. Also saves time.

Reply:
I'm resolving this conversation but it might be good reading for the other reviewers.